### PR TITLE
switch to jdk12 and maven

### PR DESCRIPTION
**Pull request by PtitNoony**

### CODE REVIEW

**Comment by PtitNoony** [comment]
Hi, I am very interrested in using your DateAxis in a personal JavaFx App.
Do not hesite to amend the pull request or ask me to make some modifications.
Thank you !

**Comment by dukke** [comment]
Hi Arnaud,

Thank you very much for taking the time to make this PR.

Have you checked it working? Does this PR work fine in JDK12?

Thanks,

**Comment by PtitNoony** [comment]
Hi,
with pleasure!
It works using jdk-12.0.1 on Windows-10, using apache-maven-3.5.2.
 Here are the pom dependency I use:

       <dependency>
            <groupId>org.openjfx</groupId>
            <artifactId>javafx-controls</artifactId>
            <version>12.0.1</version>
        </dependency>
        <dependency>
            <groupId>org.openjfx</groupId>
            <artifactId>javafx-fxml</artifactId>
            <version>12.0.1</version>
        </dependency>
        <dependency>
            <groupId>com.pixelduke</groupId>
            <artifactId>fxcharts</artifactId>
            <version>1.0-SNAPSHOT</version>
        </dependency>

**Comment by PtitNoony** [comment]
FxCharts_jdk12 0 1

**Comment by dukke** [comment]
OK, I'll just glance briefly at the code and merge if everything looks in order.

Unfortunately I don't have time to test it myself at the moment. All my spare time is currently going towards JMetro development, at the moment.

Thank you very much Arnaud!! I'll announce this new addition to FXCharts and I'll credit you for it, of course.

**Comment by dukke** [comment]
Very sorry for the delay, been very busy lately. I'll be looking at this PR very shortly (probably during the weekend).

Arnaud, thank you very much for this contribution 👍

**Comment by dukke** [comment]
First of all sorry for the delay in commenting on your PR.

All seems good @PtitNoony  (Arnaud) with the PR. Seems ready to merge.

Can you just push to a "12.0" branch instead of master? I still want to keep the java8 version for people on java8. Thank you!

**Comment by PtitNoony** [comment]
Hi, sure, but could you create a 12.0 branch please, I was not able to edit the pull request with a non existing target branch ?
Thank you !

**Comment by dukke** [comment]
Hi!

Ops sorry, that was dumb should have created the branch first :)

I've created the branch.

Would like to ask you 2 more things if that's OK with you, if not, no problem:

1 - Put the classes inside package: "com.pixelduke.chart"
2 - Would it be possible for you to use gradle instead of Maven? The problem is that I'm not very proficient with Maven and have a better knowledge of Gradle.

Thanks again!

**Comment by PtitNoony** [comment]
Hi, 
apologies for the long response time. I am not a gradle user, so I am not sure what I did I ok, please fell free to edit/comment.

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+description = 'Charts'
+
+Date buildTimeAndDate = new Date()
+ext {
+    buildDate = new SimpleDateFormat('yyyy-MM-dd').format(buildTimeAndDate)
+    buildTime = new SimpleDateFormat('HH:mm:ss.SSSZ').format(buildTimeAndDate)
+}
+
+dependencies {
+    compile 'com.googlecode.json-simple:json-simple:1.1.1'
```
Comment (dukke) [inline build.gradle]: Are you using this library?

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+repositories {
+    jcenter()
+    mavenCentral()
+}
+
+// start the demo from gradle
+task Demo(type: JavaExec) {
+    main = "com.pixelduke.chart.DemoLauncher"
```
Comment (dukke) [inline build.gradle]: This class doesn't exist

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+//    destinationDir = new File(project.buildDir, 'merged-javadoc')
+//
+//    // Note: The closures below are executed lazily.
+//    source {
+//       subprojects*.sourceSets*.main*.allSource
+//    }
+//    classpath.from {
+//        subprojects*.configurations*.compile*.copyRecursive({ !(it instanceof ProjectDependency); })*.resolve()
+//    }
+//}
```
Comment (dukke) [inline build.gradle]: Do you still need this commented out code?

File: common.gradle [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+String mavenVersion = '12.0'
+
+sourceCompatibility = '1.7'
+[compileJava, compileTestJava]*.options*.encoding = 'UTF-8'
+
+repositories {
+    mavenCentral();
+    // You may define additional repositories, or even remove "mavenCentral()".
+    // Read more about repositories here:
+    //   http://www.gradle.org/docs/current/userguide/dependency_management.html#sec:repositories
```
Comment (dukke) [inline common.gradle]: We can remove these comments

File: common.gradle [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+task createFolders(description: 'Creates the source folders if they do not exist.') doLast {
+    sourceSets*.allSource*.srcDirs*.each { File srcDir ->
+        if (!srcDir.isDirectory()) {
+            println "Creating source folder: ${srcDir}"
+            srcDir.mkdirs()
+        }
+    }
+}
```
Comment (dukke) [inline common.gradle]: Do we need this task?

File: common.gradle [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+// This file is to be applied to every subproject.
+//
+
+apply plugin: 'java'
+apply plugin: 'maven'
+
+String mavenGroupId = 'com.pixelduke'
+String mavenVersion = '12.0'
+
+sourceCompatibility = '1.7'
```
Comment (dukke) [inline common.gradle]: 12?

File: common.gradle [hunk 7]
```diff
@@ -0,0 +1,56 @@
+//
```
Comment (dukke) [inline common.gradle]: Is this build script used anywhere? I don't have actually used a common.gradle file before, reading about it, it seems that you need to put `apply from: '../common.gradle' ` to use it somewhere else?

File: settings.gradle [hunk 8]
```diff
@@ -0,0 +1,20 @@
+rootProject.name = 'chart'
+
+// Find the directories containing a "build.gradle" file in the root directory
+// of the project. That is, every directory containing a "build.gradle" will
+// be automatically the subproject of this project.
+def subDirs = rootDir.listFiles(new FileFilter() {
```
Comment (dukke) [inline settings.gradle]: I think we don't need the code starting from here all the way to the finish?

**Review by dukke — CHANGES_REQUESTED** [review]
Hi again Arnaud, 

Thanks again for your time, and for contributing back,

I've submitted a few comments to this PR.

Thanks, kind regards,

**Comment by PtitNoony** [comment]
thank you for your review. I apologize for the not clean set up, I guess I was not paying enough attention (lack of sleep).
I added a quick fix

File: settings.gradle (hunk 8 above)
Comment (dukke) [reply settings.gradle]: I think we're only missing this?

**Comment by dukke** [comment]
Hi Arnaud, 

Thanks for the changes, I think we're only missing one code review comment?

Thanks!

**Comment by PtitNoony** [comment]
Hi,
very sorry for the miss and delay...

**Review by dukke — APPROVED** [review]

**Comment by dukke** [comment]
Hi,
All seems good now?

**Comment by PtitNoony** [comment]
Hi, yes it does ! Thank you !

**Comment by dukke** [comment]
Approved and merged. Thanks Arnaud, I've tweeted about this and thank you in the tweeted for your contribution 👍

**Comment by PtitNoony** [comment]
Thank you for your patience and your kind, words !
I've been busy with a brand new little thing ! Hopefully I'll be more active in a couple of months.

**Comment by dukke** [comment]
That would be awesome!

A project I had with a client is also going to probably end soon so I'll have more time for FOSS. 👍

**Pull request merged**